src/WindowsUtils/BusyLoopUtils.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "WindowsUtils/BusyLoopUtils.h"
+
+#include <absl/base/casts.h>
+#include <absl/strings/str_format.h>
+
+#include "OrbitBase/GetLastError.h"
+#include "WindowsUtils/ReadProcessMemory.h"
+#include "WindowsUtils/SafeHandle.h"
+#include "WindowsUtils/WriteProcessMemory.h"
+
+namespace orbit_windows_utils {
+
+namespace {
+
+ErrorMessage GetLastErrorForFunction(std::string_view function) {
+  return ErrorMessage(
+      absl::StrFormat("Calling \"%s\": %s", function, orbit_base::GetLastErrorAsString()));
+}
+
+ErrorMessageOr<void> FlushInstructionCache(HANDLE process_handle, void* address, size_t size) {
+  if (::FlushInstructionCache(process_handle, address, size) == 0) {
+    return GetLastErrorForFunction("FlushInstructionCache");
+  }
+  return outcome::success();
+}
+
+ErrorMessageOr<void> WriteInstructionsAtAddress(const std::string& byte_buffer, void* address) {
+  // Mark memory page for writing.
+  DWORD old_protect = 0;
+  if (!VirtualProtect(address, byte_buffer.size(), PAGE_EXECUTE_READWRITE, &old_protect)) {
+    return GetLastErrorForFunction("VirtualProtect");
+  }
+
+  // Write original bytes.
+  memcpy(address, byte_buffer.data(), byte_buffer.size());
```

Comment:
I probably miss something: This writes into this process. Is this intended? Installing the busy loop writes into a remote process, right?

Reply:
It was intended to be used by the target process but in the end this entire function is not needed and was removed. It can be replaced by a call to "WriteProcessMemory".